code/campaignclassic/src/main/java/com/adobe/marketing/mobile/ManualCarouselTemplateNotificationBuilder.java

```diff
@@ -0,0 +1,531 @@
+        final String assetCacheLocation = CampaignPushUtils.getAssetCacheLocation();
+        final ArrayList<Bitmap> cachedImages = new ArrayList<>();
+        final ArrayList<String> imageUrls =
+                (ArrayList<String>) intentExtras.get(CampaignPushConstants.IntentKeys.IMAGE_URLS);
+        final ArrayList<String> imageCaptions =
+                (ArrayList<String>)
+                        intentExtras.get(CampaignPushConstants.IntentKeys.IMAGE_CAPTIONS);
+        final ArrayList<String> imageClickActions =
+                (ArrayList<String>)
+                        intentExtras.get(CampaignPushConstants.IntentKeys.IMAGE_CLICK_ACTIONS);
+        final String titleText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.TITLE_TEXT);
+        final String bodyText = intentExtras.getString(CampaignPushConstants.IntentKeys.BODY_TEXT);
+        final String expandedBodyText =
+                intentExtras.getString(CampaignPushConstants.IntentKeys.EXPANDED_BODY_TEXT);
+
+        if (cacheService != null && !CollectionUtils.isEmpty(imageUrls)) {
+            for (final String imageUri : imageUrls) {
+                if (!StringUtils.isNullOrEmpty(imageUri)) {
+                    final CacheResult cacheResult = cacheService.get(assetCacheLocation, imageUri);
+                    if (cacheResult != null) {
+                        cachedImages.add(BitmapFactory.decodeStream(cacheResult.getData()));
+                    }
+                }
+            }
+        }
+
+        final RemoteViews smallLayout =
+                new RemoteViews(packageName, R.layout.push_template_collapsed);
+        final RemoteViews expandedLayout =
+                new RemoteViews(packageName, R.layout.push_template_manual_carousel);
+        smallLayout.setTextViewText(R.id.notification_title, titleText);
+        smallLayout.setTextViewText(R.id.notification_body, bodyText);
+        expandedLayout.setTextViewText(R.id.notification_title, titleText);
+        expandedLayout.setTextViewText(R.id.notification_body_expanded, expandedBodyText);
+
+        final String action = intent.getAction();
+        int centerImageIndex =
+                intentExtras.getInt(CampaignPushConstants.IntentKeys.CENTER_IMAGE_INDEX);
+
+        final List<Integer> newIndices;
+        if (CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED.equals(action)) {
+            newIndices =
+                    CampaignPushUtils.calculateNewIndices(
+                            centerImageIndex,
+                            imageUrls.size(),
+                            CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED);
+        } else {
+            newIndices =
+                    CampaignPushUtils.calculateNewIndices(
+                            centerImageIndex,
+                            imageUrls.size(),
+                            CampaignPushConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED);
+        }
```

Comment:
This can be simplified as follows:
```suggestion
        final List<Integer> newIndices =. CampaignPushUtils.calculateNewIndices(centerImageIndex, imageUrls.size(), action);
```